vsutil/__init__.py

```diff
@@ -1,20 +1,24 @@
 """
 VSUtil. A collection of general-purpose VapourSynth functions to be reused in modules and scripts.
 """
-__all__ = ['Dither', 'Range', 'depth', 'fallback', 'frame2clip', 'get_depth', 'get_plane_size',
-           'get_subsampling', 'get_w', 'get_y', 'insert_clip', 'is_image', 'iterate', 'join', 'plane', 'split']
+__all__ = ['Dither', 'Range', 'depth', 'disallow_variable_format', 'disallow_variable_resolution', 'fallback',
+           'frame2clip', 'get_depth', 'get_plane_size',
+           'get_subsampling', 'get_w', 'get_y', 'insert_clip', 'is_image', 'iterate', 'join', 'plane',
+           'split']
 
 from enum import Enum, IntEnum
+from functools import wraps
 from mimetypes import types_map
 from os import path
-from typing import Any, Callable, List, Literal, Optional, Tuple, Type, TypeVar, Union
+from typing import Any, Callable, cast, List, Literal, Optional, Tuple, Type, TypeVar, Union
 
 import vapoursynth as vs
 core = vs.core
 
 T = TypeVar('T')
 E = TypeVar('E', bound=Enum)
 R = TypeVar('R')
+F = TypeVar('F', bound=Callable[..., Any])
```

Comment:
Can you try to mark this as covariant.

Reply:
```
__init__.py:42: error: Cannot use a covariant type variable as a parameter
__init__.py:55: error: Cannot use a covariant type variable as a parameter
```
This is what I get after doing that, whats the reason behind this change anyways?

Reply:
covariant marks a parameter as "accepts subtypes too".

Reply:
![image](https://user-images.githubusercontent.com/10972040/84077660-af104b00-a9a5-11ea-9579-5ff827129766.png)
Doesn't `bound` in this case accomplish what we want?

Reply:
I see. I think it was worth a try though.